python/xorq/vendor/ibis/expr/types/core.py

```diff
@@ -388,20 +396,33 @@ def _find_backend_original(self, *, use_default: bool = False) -> BaseBackend:
 
     def _find_backend(self, *, use_default=True):
         from xorq.config import _backend_init
-        from xorq.expr.relations import CachedNode, Read
 
         try:
-            if tuple(op.source for op in self.op().find((Read, CachedNode))):
-                current_backend = _backend_init()
-            else:
-                current_backend = self._find_backend_original(use_default=use_default)
+            current_backend = self._find_backend_original(use_default=use_default)
         except XorqError as e:
             if "Multiple backends found" in e.args[0]:
                 current_backend = _backend_init()
             else:
                 raise e
         return current_backend
 
+    def _find_backend_non_tables(self):
+        from xorq.config import _backend_init
+        from xorq.expr.relations import CachedNode, Read
+
+        node_types = (Read, CachedNode)
+        backends = set()
+        for node in self.op().find(node_types):
+            backends.add(node.source)
```

Comment:
this doesn't work because of how we define `BaseBackend.__hash__`
```
In [1]: import xorq as xo

In [2]: len(set(xo.connect() for _ in range(10)))
Out[2]: 1

In [3]: len(set(xo.connect()._profile for _ in range(10)))
Out[3]: 10
```